components/org.wso2.carbon.identity.query.saml/src/test/java/org/wso2/carbon/identity/query/saml/QueryResponseBuilderTest.java

```diff
@@ -0,0 +1,187 @@
+
+/**
+ * Test Class for the QueryResponseBuilder
+ */
+@PrepareForTest({OpenSAML3Util.class, QueryResponseBuilder.class})
+public class QueryResponseBuilderTest extends PowerMockTestCase {
+
+    @Test
+    public void testBuildforSuccess() throws Exception {
+
+        DummyAssertion dummyAssertion = new DummyAssertion();
+        List<Assertion> assertions = new ArrayList<>();
+        SAMLSSOServiceProviderDO ssoIdpConfigs = new SAMLSSOServiceProviderDO();
+        Response response = new DummyResponse();
+        assertions.add(dummyAssertion);
+        DummyIssuer issuer = new DummyIssuer();
+
+        mockStatic(OpenSAML3Util.class);
+        when(OpenSAML3Util.getIssuer(anyString())).thenReturn(issuer);
+        when(OpenSAML3Util.setSignature(any(Response.class), anyString(), anyString(), any(SignKeyDataHolder.class))).thenReturn(response);
+
+        SignKeyDataHolder testSign = Mockito.mock(SignKeyDataHolder.class);
+        whenNew(SignKeyDataHolder.class).withAnyArguments().thenReturn(testSign);
+        assertTrue(QueryResponseBuilder.build(assertions, ssoIdpConfigs, "test").getAssertions() != null);
+
+    }
+
+    @Test
+    public void testBuildforError() throws IdentitySAML2QueryException {
+
+        DummyIssuer issuer = new DummyIssuer();
+        DummyIssuer issuer2 = new DummyIssuer();
+        List<InvalidItemDTO> invalidItems = new ArrayList<>();
+        mockStatic(OpenSAML3Util.class);
+        when(OpenSAML3Util.getIssuer(anyString())).thenReturn(issuer);
+        Response testresponse1 = QueryResponseBuilder.build(invalidItems);
+        when(OpenSAML3Util.getIssuer(anyString())).thenReturn(issuer2);
+        invalidItems.add(new InvalidItemDTO(SAMLQueryRequestConstants.ValidationType.VAL_SUBJECT,
+                SAMLQueryRequestConstants.ValidationMessage.VAL_SUBJECT_ERROR));
+        Response testresponse2 = QueryResponseBuilder.build(invalidItems);
+
+        assertEquals(testresponse1.getStatus().getStatusCode().getValue(), null);
+        assertEquals(testresponse1.getStatus().getStatusMessage().getMessage(), null);
+        assertEquals(testresponse2.getStatus().getStatusMessage().getMessage(), "Request subject is invalid");
+        assertEquals(testresponse2.getStatus().getStatusCode().getValue(), "urn:oasis:names:tc:SAML:2.0:status:Requester");
+    }
+
+    @Test
+    public void testBuildStatus() {
+
+        Status dummyStatus1 = QueryResponseBuilder.buildStatus("teststatus1", "testmsg");
+        Status dummyStatus2 = QueryResponseBuilder.buildStatus("teststatus2", null);
+        assertEquals(dummyStatus1.getStatusMessage().getMessage(), "testmsg");
+        assertEquals(dummyStatus2.getStatusMessage(), null);
+        assertEquals(dummyStatus1.getStatusCode().getValue(), "teststatus1");
+        assertEquals(dummyStatus2.getStatusCode().getValue(), "teststatus2");
+    }
+
+    @DataProvider(name = "provideStatusCode")
+    public Object[][] createSubject() {
```

Comment:
This does not seem like a good method name for a dataProvider.

Reply:
Okay, Ill change it